src/weather_module.js

```diff
@@ -0,0 +1,33 @@
+import {
+  celciusDisplay,
+  fahrenheitDisplay,
+  displayError,
+  changeWeatherImg,
+  flashInfo,
+} from './dom_module';
+
+export default function getWeatherInfos(location, units = 'metric') {
```

Comment:
**[OPTIONAL but actually it is necessary, but I don't wanna add this anymore as required]**

Data fetching should be its own function . As it is right now, the `getWeatherInfos` function is violating the single responsibility rule. It fetches data as well as processing that data. I would have created a separate function that fetches data only:

```javascript
const fetchWeatherData(city) => {
  // API key should be private but ok let's stick it here
  const apiKey = 617ec88a7d93c5b6ace5a90f054e9838;
  // i would by default set the units to metric and not have to pass it as argument
 // i would then later on just let my toggle button do the converting 
 const url = `https://api.openweathermap.org/data/2.5/weather?q=${city}&APPID=${apiKey}&metric

 return fetch(url, {mode: 'cors'})
               .then((res) => res.json())
               .then((data) => data)
               .catch((err) => err)
} 
```

In another function, in another file (maybe called `utils.js`) I would have placed there my logic for temperature conversion:

```javascript
const convertTemp = (temp, currentUnit) => {
  if(currentUnit === 'C'){
    return 9/5(temp) + 32
  }else if(currentUnit === 'F'){
   return 5/9(temp - 32);
 }
} 
```

Further, I'd have another module/function which will process the data gathered from `fetchData` above:

```javascript
const processData = (city) => {
  fetchData(city)
    .then((data) => {
       // do what we need to do when the data fetching is successful
       }
     )
    .catch(err => { // let's do error handling }
}
```

Still further, I would have used `async await`  as well as `try` and `catch` instead of a series of `then`s. For instance in the `fetchData` function I set as an example above:

```javascript
const fetchData = async (city) => {
 const apiKey = // <same API key above>
 const url = // same URL above
 
 try {
   const res = await fetch(url, {mode: 'cors'});
   return res.json();
 }catch((err) => {
  return err;
 })
}
```

Then whatever the result of calling `fetchData` we can now set into a variable without having to treat it as a `promise`:

```javascript
 const city = "Tokyo";
const data = fetchData(city); // can either return the json data or an error data
```